### PR TITLE
Fix authentication options binding

### PR DESCRIPTION
**Pull request by zpqrtbnk**
See related issue.

### CODE REVIEW

File: src/Hazelcast.Net.Tests/Configuration/HazelcastOptionsTests.cs [hunk 1]
```diff
@@ -235,6 +235,105 @@ public void AuthenticationOptionsFile()
...
 
+        [Test]
+        public void AuthenticationUsernamePassword()
+        {
+            const string json = @"{ ""hazelcast"": {
+""authentication"" : {
+    ""username-password"": { ""username"": ""bob"", ""password"": ""secret"" }
+}
+}}";
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes(json));
```
Comment (emreyigit) [inline src/Hazelcast.Net.Tests/Configuration/HazelcastOptionsTests.cs:246]: Could you consider to use `ReadResource(json)` for name of not to repeat parsing part? We have already a method to do that just above.

**Comment by emreyigit** [comment]
Just left one small comment. Other than that, looks great.

**Review by emreyigit — DISMISSED** [review]

**Review by alisengul53 — COMMENTED** [review]
Do we have any integration test which is using a real server(maybe with RC) for these authentication types?
Or maybe we also don't have for other clients. I have no idea

File: src/Hazelcast.Net.Tests/Configuration/HazelcastOptionsTests.cs (hunk 1 above)
Comment (zpqrtbnk) [reply src/Hazelcast.Net.Tests/Configuration/HazelcastOptionsTests.cs]: Ah sure, have pushed a commit.

**Comment by zpqrtbnk** [comment]
Ali: we don't have such tests as far as I am aware. OTOH, these tests here only validate that we parse the options correctly.

**Review by alisengul53 — DISMISSED** [review]
LGTM

**Review by emreyigit — APPROVED** [review]

**Review by alisengul53 — APPROVED** [review]

**Pull request merged**